internal/migrate/state.go

```diff
@@ -0,0 +1,136 @@
+/*
+ * Flow CLI
+ *
+ * Copyright 2019 Dapper Labs, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package migrate
+
+import (
+	"fmt"
+	"os"
+
+	"github.com/onflow/cadence/runtime/common"
+	"github.com/onflow/flow-emulator/storage/migration"
+	emulatorMigrate "github.com/onflow/flow-emulator/storage/migration"
+	"github.com/onflow/flow-emulator/storage/sqlite"
+	"github.com/onflow/flow-go-sdk"
+	"github.com/onflow/flow-go/cmd/util/ledger/migrations"
+	"github.com/onflow/flow-go/cmd/util/ledger/reporters"
+	"github.com/onflow/flowkit/v2"
+	"github.com/onflow/flowkit/v2/config"
+	"github.com/onflow/flowkit/v2/output"
+	"github.com/rs/zerolog"
+	"github.com/spf13/cobra"
+
+	"github.com/onflow/flow-cli/internal/command"
+)
+
+var stateFlags struct {
+	Contracts  []string `default:"" flag:"contracts" info:"contract names to migrate"`
+	SaveReport string   `default:"" flag:"save-report" info:"save migration report to a given directory if provided"`
```

Comment:
Added a `--save-report` flag - open to feedback here about naming/acceptable defaults.  Assumed it's probably best not to pollute user's workspace if not manually specified (so default to NOOP) but can change if wanted.

cc @turbolent